companion/src/simulation/telemetrysimu.cpp

```diff
@@ -154,6 +157,49 @@ void TelemetrySimulator::onLogTimerEvent()
   logPlayback->stepForward(false);
 }
 
+void TelemetrySimulator::onGpsRunLoop()
+{
+  int a = ui->gps_latlon->text().contains(",");
+  if (!a) {
+    QMessageBox::information(this,tr("Bad GPS Format"),tr("Must be decimal latitude,longitude"));
+    ui->gps_latlon->setText("000.00000000,000.00000000");
+    ui->GPSpushButton->click();
+  }
+  else
+  {
+    QStringList lalo = (ui->gps_latlon->text()).split(",");
+    QString lat = lalo[0];
+    QString lon = lalo[1];
+    double B2 = lat.toDouble();
```

Comment:
camel case for variables please. My bad for not advising earlier.

Reply:
Sorry. I do not know what camel case is. Can you give me an example?

Reply:
lowercase first letter and caps for intermediate eg b2 or isKeyPressed

Reply:
I was editing my comment at the same time you were answering. Should I change variables such as lalo to laLo? What about 3 letter variables like lat and lon?

Reply:
The simple answer 1st, 3 letter variables all lower case as they represent a single word.
In the case of the example of lalo technically yes laLo however it is a temporary variable with a life of 3 sequential lines so can take the all lower case form. I personally would name it in relation to its source eg gpsLatLon which then flows to its use.
However you don't need lines 171 & 172 as the convert to double can be direct from the strings eg double b2 = latlon(0),toDouble();